apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/ConfigsExportService.java

```diff
@@ -0,0 +1,187 @@
+   */
+  private static void writeAsZipOutputStream(Stream<ConfigBO> configBOStream, OutputStream outputStream)
+      throws IOException {
+    try (final ZipOutputStream zipOutputStream = new ZipOutputStream(outputStream)) {
+      final Consumer<ConfigBO> configBOConsumer = configBO -> {
+        try {
+          // TODO, Stream.reduce will cause some problems. Is There other way to speed up the downloading?
+          synchronized (zipOutputStream) {
+            write2ZipOutputStream(zipOutputStream, configBO);
+          }
+        } catch (IOException e) {
+          logger.error("Write error. {}", configBO);
+          throw new IllegalStateException(e);
+        }
+      };
+      configBOStream.forEach(configBOConsumer);
+    }
+  }
+
+  /**
+   * write {@link ConfigBO} as file to {@link ZipOutputStream}.
+   * Watch out the concurrent problem!
+   * zip output stream is same like cannot write concurrently!
+   * the name of file is determined by {@link ConfigFileUtils#toFilename(String, String, String, ConfigFileFormat)}.
+   * the path of file is determined by {@link ConfigFileUtils#toFilePath(String, String, Env, String)}.
+   * @param zipOutputStream zip file output stream
+   * @param configBO a namespace represent
+   * @return zip file output stream same as parameter zipOutputStream
+   */
+  private static ZipOutputStream write2ZipOutputStream(
+      final ZipOutputStream zipOutputStream,
+      final ConfigBO configBO
+  ) throws IOException {
+    final Env env = configBO.getEnv();
+    final String ownerName = configBO.getOwnerName();
+    final String appId = configBO.getAppId();
+    final String clusterName = configBO.getClusterName();
+    final String namespace = configBO.getNamespace();
+    final String configFileContent = configBO.getConfigFileContent();
+    final ConfigFileFormat configFileFormat = configBO.getFormat();
+    final String configFilename = ConfigFileUtils.toFilename(appId, clusterName, namespace, configFileFormat);
+    final String filePath = ConfigFileUtils.toFilePath(ownerName, appId, env, configFilename);
+    final ZipEntry zipEntry = new ZipEntry(filePath);
+    try {
+      zipOutputStream.putNextEntry(zipEntry);
+      zipOutputStream.write(configFileContent.getBytes());
+      zipOutputStream.closeEntry();
+    } catch (IOException e) {
+      logger.error("config export failed. {}", configBO);
+      throw new IOException("config export failed", e);
+    }
+    return zipOutputStream;
+  }
+
+  /**
+   * @return the namespaces current user exists
+   */
+  private Stream<ConfigBO> makeStreamBy(
+      final Env env,
+      final String ownerName,
+      final String appId,
+      final String clusterName
+  ) {
+    final List<NamespaceBO> namespaceBOS = namespaceService.findNamespaceBOs(appId, env, clusterName);
+    final Function<NamespaceBO, ConfigBO> function = namespaceBO -> new ConfigBO(env, ownerName, appId, clusterName, namespaceBO);
+    return namespaceBOS.parallelStream()
```

Comment:
please format all the changed codes with [code style](https://github.com/ctripcorp/apollo/tree/master/apollo-buildtools/style)